Src/DependencyCollector/Shared/Implementation/TelemetryDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,186 @@
+
+
+        public TelemetryDiagnosticSourceListener(TelemetryConfiguration configuration, ICollection<string> excludeDiagnosticSourceActivities) 
+            : base(configuration)
+        {
+            this.Client.Context.GetInternalContext().SdkVersion = SdkVersionUtils.GetSdkVersion("rdd" + RddSource.DiagnosticSourceListener + ":");
+            this.PrepareExclusionLists(excludeDiagnosticSourceActivities);
+        }
+
+        internal override bool IsSourceEnabled(DiagnosticListener value)
+        {
+            return !this.excludedDiagnosticSources.Contains(value.Name);
+        }
+
+        protected override HashSet<string> GetListenerContext(DiagnosticListener diagnosticListener)
+        {
+            HashSet<string> excludedActivities;
+            if (!this.excludedDiagnosticSourceActivities.TryGetValue(diagnosticListener.Name, out excludedActivities))
+            {
+                return null;
+            }
+
+            return excludedActivities;
+        }
+
+        internal override bool IsEventEnabled(string evnt, object input1, object input2, DiagnosticListener diagnosticListener, HashSet<string> context)
+        {
+            return !this.IsActivityExcluded(evnt, context)
+                && !evnt.EndsWith(ActivityStartNameSuffix, StringComparison.OrdinalIgnoreCase);
+        }
+
+        internal override bool ShouldHandleEvent(KeyValuePair<string, object> evnt, DiagnosticListener diagnosticListener, HashSet<string> context)
+        {
+            return !this.IsActivityExcluded(evnt.Key, context)
+                && evnt.Key.EndsWith(ActivityStopNameSuffix, StringComparison.OrdinalIgnoreCase);
+        }
+
+        internal bool IsActivityExcluded(string activityName, HashSet<string> excludedActivities)
+        {
+            return excludedActivities?.Contains(activityName) == true;
+        }
+
+        internal override void HandleEvent(KeyValuePair<string, object> evnt, DiagnosticListener diagnosticListener, HashSet<string> context)
+        {
+            Activity currentActivity = Activity.Current;
+            if (currentActivity == null)
+            {
+                DependencyCollectorEventSource.Log.CurrentActivityIsNull();
+                return;
+            }
+
+            DependencyCollectorEventSource.Log.TelemetryDiagnosticSourceListenerActivityStopped(currentActivity.Id, currentActivity.OperationName);
+
+            // extensibility point - can chain more telemetry extraction methods here
+            ITelemetry telemetry = this.ExtractDependencyTelemetry(diagnosticListener, currentActivity);
+            if (telemetry == null)
+            {
+                return;
+            }
+
+            // properly fill dependency telemetry operation context
+            telemetry.Context.Operation.Id = currentActivity.RootId;
+            telemetry.Context.Operation.ParentId = currentActivity.ParentId;
+            telemetry.Timestamp = currentActivity.StartTimeUtc;
+
```

Comment:
duration

Reply:
```Duration``` is specific to RDD telemetry and it is initialized inside ```ExtractDependencyTelemetry()``` method

Reply:
ok, thanks for explaination